File: atividade 6 java/IPCA.java
import java.util.Scanner;

public class IPCA {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        double menor = -1, maior = -1, soma = 0;
        String mesAno, menorMesAno = "", maiorMesAno = "";
        int count = 0;

        System.out.println("Digite os dados do IPCA (ano/mês índice). Para encerrar, digite *:");

        while (true) {
            String entrada = scanner.nextLine();

            if (entrada.equals("*")) {
                break;
            }

            String[] partes = entrada.split(" ");
            mesAno = partes[0];
            double ipcavalor = Double.parseDouble(partes[1].replace(",", "."));

            if (count == 0 || ipcavalor < menor) {
                menor = ipcavalor;
                menorMesAno = mesAno;
            }

            if (count == 0 || ipcavalor > maior) {
                maior = ipcavalor;
                maiorMesAno = mesAno;
            }

            soma += ipcavalor;
            count++;
        }

        if (count > 0) {
            System.out.printf("Menor IPCA: %.2f (Mês/Ano: %s)\n", menor, menorMesAno);
            System.out.printf("Maior IPCA: %.2f (Mês/Ano: %s)\n", maior, maiorMesAno);
            System.out.printf("Média do IPCA: %.2f\n", soma / count);
        } else {
            System.out.println("Nenhum dado fornecido.");
        }

        scanner.close();
    }
}

File: atividade 6 c/ipca.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    char entrada[50];
    double ipcavalor, menor = -1, maior = -1, soma = 0;
    char mes_ano[8], menor_mes_ano[8], maior_mes_ano[8];
    int primeiro = 1, count = 0;

    printf("Digite os dados do IPCA (ano/mês índice). Para encerrar, digite *:\n");

    while (1) {
        fgets(entrada, sizeof(entrada), stdin);

        if (entrada[0] == '*') {
            break;
        }

        sscanf(entrada, "%s %lf", mes_ano, &ipcavalor);

        if (primeiro || ipcavalor < menor) {
            menor = ipcavalor;
            sprintf(menor_mes_ano, "%s", mes_ano);
        }

        if (primeiro || ipcavalor > maior) {
            maior = ipcavalor;
            sprintf(maior_mes_ano, "%s", mes_ano);
        }

        soma += ipcavalor;
        count++;
        primeiro = 0;
    }

    if (count > 0) {
        printf("Menor IPCA: %.2lf (%s)\n", menor, menor_mes_ano);
        printf("Maior IPCA: %.2lf (%s)\n", maior, maior_mes_ano);
        printf("Média do IPCA: %.2lf\n", soma / count);
    } else {
        printf("Nenhum dado fornecido.\n");
    }

    return 0;
}
